enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1639,7 +1639,9 @@ func (p *PebbleCache) Get(ctx context.Context, r *rspb.ResourceName) ([]byte, er
 		return nil, err
 	}
 	defer rc.Close()
-	return io.ReadAll(rc)
+	var buffer bytes.Buffer
```

Comment:
why don't we allocate the full buffer up front? presumably we know the size of the contents in the cache.

Reply:
When I pre-allocate the buffer, the performance seems worse:

```
                            │ /home/lulu/baseline_readall_nocdc.log │ /home/lulu/benchmark_copy_nocdc.log │ /home/lulu/benchmark_copy_nocdc-pre-allocate-make.log │
                             │                sec/op                 │    sec/op     vs base               │            sec/op              vs base                │
Get/Pebble/size=10-24                                    8.290µ ± 5%   9.132µ ±  3%  +10.16% (p=0.002 n=6)                    9.287µ ±  4%   +12.03% (p=0.002 n=6)
Get/Pebble/size=100-24                                   8.753µ ± 4%   9.545µ ±  5%   +9.05% (p=0.004 n=6)                    9.820µ ±  6%   +12.20% (p=0.002 n=6)
Get/Pebble/size=1000-24                                  8.889µ ± 4%   9.777µ ±  6%  +10.00% (p=0.002 n=6)                    9.787µ ±  4%   +10.10% (p=0.002 n=6)
Get/Pebble/size=10000-24                                 16.22µ ± 4%   17.14µ ±  8%   +5.68% (p=0.026 n=6)                    21.81µ ±  8%   +34.45% (p=0.002 n=6)
Get/Pebble/size=100000-24                                19.23µ ± 2%   19.85µ ±  3%   +3.25% (p=0.009 n=6)                    34.53µ ±  3%   +79.56% (p=0.002 n=6)
Get/Pebble/size=1000000-24                               31.98µ ± 2%   25.31µ ±  4%  -20.88% (p=0.002 n=6)                    81.38µ ±  2%  +154.44% (p=0.002 n=6)
Get/Pebble/size=10000000-24                             123.77µ ± 8%   79.70µ ±  7%  -35.61% (p=0.002 n=6)                   535.68µ ± 10%  +332.82% (p=0.002 n=6)
Get/Pebble/size=50000000-24                              531.5µ ± 6%   261.4µ ± 11%  -50.82% (p=0.002 n=6)                   2729.2µ ±  6%  +413.50% (p=0.002 n=6)
Get/Pebble/size=100000000-24                             902.8µ ± 3%   472.8µ ±  2%  -47.63% (p=0.002 n=6)                   5539.8µ ±  4%  +513.63% (p=0.002 n=6)
geomean                                                  41.71µ        34.67µ        -16.89%                                  91.35µ        +119.01%

```